include/envoy/server/filter_config.h

```diff
@@ -90,11 +90,12 @@ class ProtocolOptionsFactory : public Config::TypedFactory {
    * @param validation_visitor message validation visitor instance.
    * @return Upstream::ProtocolOptionsConfigConstSharedPtr the protocol options
    */
-  virtual Upstream::ProtocolOptionsConfigConstSharedPtr
-  createProtocolOptionsConfig(const Protobuf::Message& config,
-                              ProtobufMessage::ValidationVisitor& validation_visitor) {
+  virtual Upstream::ProtocolOptionsConfigConstSharedPtr createProtocolOptionsConfig(
+      const Protobuf::Message& config, ProtobufMessage::ValidationVisitor& validation_visitor,
+      Server::Configuration::TransportSocketFactoryContext& factory_context) {
```

Comment:
nit: why do we need to pass validation_visitor separately if it is available from factory_context? I think same applies to ClusterInfoImpl, so it's been added awhile ago.

Reply:
good point, as it was passed in along with the factory_context in the code i refactored, i didn't realized it is part of it.